test/cookie-sessions.test.js

```diff
@@ -17,104 +20,100 @@ describe('cookie sessions tests', () => {
+      assert.ok(iv)
+      assert.ok(authTag)
     })
 
-    it('should return error on long data', (done) => {
+    it('should return error on long data', function() {
       var  long = '';
-      for(var i=0; i<2050; i++){
+      for(var i = 0; i < 1500; i++){
           long += 'a';
       };
-      sandbox.stub(sessions, 'encrypt').yields(null, {ciphertext: long, nonce: long});
-      sessions.serialize(secret, user_data, (err, result) => {
-        assert.ok(err)
+      this.sandbox.stub(sessions, 'encrypt').returns({ciphertext: long, iv: long, authTag: long});
+      try {
+        var cookiestr = sessions.serialize(secret, user_data)
+      }
+      catch(err) {
         assert.equal(err.message, 'Data too long to store in a cookie')
-        done();
-      });
+      };
     })
   })
 
-  describe('deserialize', () => {
-    it('should return user data in JSON if cookie has not expired', (done) => {
-      sessions.serialize(secret, user_data, (err, result) => {
-        sessions.deserialize(secret, result, 1000 * 60 * 60 * 24, (err, data) => {
-          assert.ok(!err)
-          assert.deepEqual(data, user_data)
-          done();
-        });
-      });
+  describe('deserialize', function() {
+    it('should return user data in JSON if cookie has not expired', function() {
+      var cookiestr = sessions.serialize(secret, user_data, Date.now(), false)
+      var jsonData = sessions.deserialize(secret, cookiestr, 1000 * 60 * 60 * 12)
+      assert.deepEqual(jsonData.data, user_data)
     })
 
-    it('should return validation error if cookie has expired', (done) => {
-      var str = 'b5a8af11ce10cd632796df523c9142595b472b686738f6a6$29d28d5d9c63b3daf6d628626b5ce3d045dc506321579af014b7eda3665e84342f65c97f79df314b7991255a41585e325e7a6e7e0218485c27af644cc1517e81801dae85716deeea32fbdccebda1ed59a3b3a1ec689447d5a39d6a37c4541623ef93ffc0ca2fb17cd20e4708a7fab59e3f87c5c5afa277b66bd3'
-      sessions.deserialize(secret, str, 1000 * 60 * 60 * 24, (err, data) => {
+    it('should return validation error if cookie has expired', function() {
+      var str = '3d11d4581d2359247a4183fc3c84aceaa5835d38075117b10c1f8bcff4a63f0d3740f6ff30f60942b9851af38fd8f1d5bcf6b2376c6803f001a844dc71b5d50276cbb735f501c47d320ef6c6d07cd2047b7a6b26521c23687718ebf248bd9b$69d2b05d79b47c3e5a94b0f32f2fbe3b$ed4c06a57df5a2b646f7e5f82bffb17f'
+      try {
+        var jsonData = sessions.deserialize(secret, str, 1000 * 60 * 60 * 10)
+      }
+      catch(err) {
         assert.ok(err)
         assert.equal(err.message, 'Cookie expired')
-        done();
-      });
+      }
     })
   })
 
-  describe('readSession', () => {
-    it('session with key node_session read ok', () => {
-      sandbox.stub(sessions, 'deserialize').yields(null, {username: 'foobar'});
-      sessions.readSession('node_session', secret, 12, {cookies: {node_session: 'lala'}} , (err, data) => {
-        assert.ok(!err)
-        assert.deepEqual({username: 'foobar'}, data)
-      })
+  /*
```

Comment:
why did we comment this out?  If it's not needed, we should just remove it.

Reply:
I removing it. I forgot to do so. There's no readSession function now.